scanpy/preprocessing/_normalization.py

```diff
@@ -27,6 +32,217 @@ def _normalize_data(X, counts, after=None, copy=False):
     return X
 
 
+def _pearson_residuals(X, theta, clip, copy=False):
+
+    X = X.copy() if copy else X
+
+    # check theta
+    if theta <= 0:
+        # TODO: would "underdispersion" with negative theta make sense?
+        # then only theta=0 were undefined..
+        raise ValueError('Pearson residuals require theta > 0')
+    # prepare clipping
+    if clip == 'auto':
+        n = X.shape[0]
+        clip = np.sqrt(n)
+    if clip < 0:
+        raise ValueError("Pearson residuals require `clip>=0` or `clip='auto'`.")
+
+    if check_nonnegative_integers(X) is False:
+        raise ValueError("`pp.normalize_pearson_residuals` expects raw count data")
+
+    if sp_sparse.issparse(X):
+        sums_genes = np.sum(X, axis=0)
+        sums_cells = np.sum(X, axis=1)
+        sum_total = np.sum(sums_genes).squeeze()
+    else:
+        sums_genes = np.sum(X, axis=0, keepdims=True)
+        sums_cells = np.sum(X, axis=1, keepdims=True)
+        sum_total = np.sum(sums_genes)
+
+    mu = np.array(sums_cells @ sums_genes / sum_total)
+    diff = np.array(X - mu)
+    residuals = diff / np.sqrt(mu + mu ** 2 / theta)
+
+    # clip
+    residuals = np.clip(residuals, a_min=-clip, a_max=clip)
+
+    return residuals
+
+
+def normalize_pearson_residuals(
+    adata: AnnData,
+    theta: float = 100,
+    clip: Union[Literal['auto', 'none'], float] = 'auto',
+    layer: Optional[str] = None,
+    copy: bool = False,
+    inplace: bool = True,
+) -> Optional[Dict[str, np.ndarray]]:
+    """\
+    Computes analytic Pearson residuals, assuming a negative binomial offset
+    model with overdispersion theta shared across genes. By default, residuals
+    are clipped to sqrt(n) and overdispersion theta=100 is used.
```

Comment:
think would be cool to add reference to your work (bioarxiv?)

Reply:
good idea.
I added an entry to the `references.rst` and added the reference to the relevant docstrings!